static/js/publisher/form/multiselect.js

```diff
@@ -0,0 +1,376 @@
+   * Remove it from the 'values' list.
+   */
+  addItem(e) {
+    // Get the key from the 'data-key' attribute of the element
+    const key = e.target.dataset.key;
+    // Get the object based on the key
+    const toAdd = this.state.values.filter(item => item.key === key)[0];
+    const newSelected = this.state.selected.slice(0);
+    newSelected.push(toAdd);
+    const newValues = this.state.values.filter(item => item.key !== key);
+    this.searchInput.value = '';
+
+    this.setState({
+      values: newValues,
+      selected: newSelected,
+      searchTerm: '',
+      // Preserve the filtered list in the dropdown
+      searchResults: newValues
+    }, this.updateInputValue);
+  }
+
+  /**
+   * Filter a list by term
+   *
+   * @param values
+   * @param searchTerm
+   * @returns {{name: string, key: string}[]}
+   */
+  filterByTerm(values, searchTerm) {
+    searchTerm = searchTerm || this.state.searchTerm;
+    return values.filter(item => item.name.toLowerCase().startsWith(searchTerm.toLowerCase()));
+  }
+
+  /**
+   * Filter the 'values' list by the search term.
+   */
+  search(e) {
+    const searchTerm = e.target.value;
+    const searchResults = this.filterByTerm(this.state.values, searchTerm);
+    let highlighted = this.state.highlightedOption;
+    if (highlighted) {
+      if (highlighted > searchResults.length - 1) {
+        highlighted = searchResults.length - 1;
+      }
+      if (highlighted < 0) {
+        highlighted = 0;
+      }
+    }
+
+    this.setState({
+      searchResults: searchResults,
+      searchTerm: searchTerm,
+      showSearch: true,
+      highlightedOption: highlighted
+    });
+  }
+
+  /**
+   * Update the original input value and dispatch a change event to the input and form.
+   */
+  updateInputValue() {
+    this.props.input.setAttribute('value', this.state.selected.map(item => item.key).join(', '));
+    const changeEvent = new Event('change');
+    const form = this.props.input.closest('form');
+
+    this.props.input.dispatchEvent(changeEvent);
+    if (form) {
+      form.dispatchEvent(changeEvent);
+    }
+  }
+
+  /**
+   * Focus on the input
+   */
+  focusInput() {
+    this.searchInput.focus();
+    this.setState({
+      showSearch: true
+    });
+  }
+
+  /**
+   * Navigate the component with the keyboard.
+   *
+   * @param event
+   */
+  handleKeypress(event) {
+    if (this.state.showSearch) {
+      let highlighted = this.state.highlightedOption;
+      let results = this.state.searchResults;
+
+      switch (event.key) {
+        case 'ArrowDown':
+          if (highlighted === null) {
+            highlighted = 0;
+          } else {
+            highlighted += 1;
+          }
+          break;
+        case 'ArrowUp':
+          if (highlighted > 0) {
+            highlighted -= 1;
+          }
+          break;
+        case 'Enter':
+          event.preventDefault();
+          if (highlighted >= 0) {
+            this.addItem({
+              target: {
```

Comment:
See the comment about `removeItem`. Please separate `addItem(key)` and `addItemClick(event)` functions not to duct tape fake event object.